tests/integration/helpers/cluster.py

```diff
@@ -184,36 +215,294 @@ def __init__(self, base_path, name=None, base_config_dir=None, server_bin_path=N
+        self.hdfs_dir = p.abspath(p.join(self.instances_dir, "hdfs"))
+        self.hdfs_logs_dir = os.path.join(self.hdfs_dir, "logs")
+
+        # available when with_kerberized_hdfs == True
+        self.hdfs_kerberized_host = "kerberizedhdfs1"
+        self.hdfs_kerberized_name_port = get_open_port()
+        self.hdfs_kerberized_data_port = get_open_port()
+        self.hdfs_kerberized_dir = p.abspath(p.join(self.instances_dir, "kerberized_hdfs"))
+        self.hdfs_kerberized_logs_dir = os.path.join(self.hdfs_kerberized_dir, "logs")
+
         # available when with_kafka == True
-        self.schema_registry_client = None
+        self.kafka_host = "kafka1"
+        self.kafka_port = get_open_port()
+        self.kafka_docker_id = None
         self.schema_registry_host = "schema-registry"
-        self.schema_registry_port = 8081
+        self.schema_registry_port = get_open_port()
+        self.kafka_docker_id = self.get_instance_docker_id(self.kafka_host)
+
+        # available when with_kerberozed_kafka == True
+        self.kerberized_kafka_host = "kerberized_kafka1"
+        self.kerberized_kafka_port = get_open_port()
+        self.kerberized_kafka_docker_id = self.get_instance_docker_id(self.kerberized_kafka_host)
+
+        # available when with_mongo == True
+        self.mongo_host = "mongo1"
+        self.mongo_port = get_open_port()
+
+        # available when with_cassandra == True
+        self.cassandra_host = "cassandra1"
+        self.cassandra_port = 9042
+        self.cassandra_ip = None
+        self.cassandra_id = self.get_instance_docker_id(self.cassandra_host)
+
+        # available when with_rabbitmq == True
+        self.rabbitmq_host = "rabbitmq1"
+        self.rabbitmq_port = get_open_port()
+        self.rabbitmq_http_port = get_open_port()
+
+        # available when with_redis == True
+        self.redis_host = "redis1"
+        self.redis_port = get_open_port()
+
+        # available when with_postgres == True
+        self.postgres_host = "postgres1"
+        self.postgres_ip = None
+        self.postgres2_host = "postgres2"
+        self.postgres2_ip = None
+        self.postgres3_host = "postgres3"
+        self.postgres3_ip = None
+        self.postgres4_host = "postgres4"
+        self.postgres4_ip = None
+        self.postgres_port = 5432
+        self.postgres_dir = p.abspath(p.join(self.instances_dir, "postgres"))
+        self.postgres_logs_dir = os.path.join(self.postgres_dir, "postgres1")
+        self.postgres2_logs_dir = os.path.join(self.postgres_dir, "postgres2")
+        self.postgres3_logs_dir = os.path.join(self.postgres_dir, "postgres3")
+        self.postgres4_logs_dir = os.path.join(self.postgres_dir, "postgres4")
+
+        # available when with_mysql_client == True
+        self.mysql_client_host = "mysql_client"
+        self.mysql_client_container = None
+ 
+        # available when with_mysql == True
+        self.mysql_host = "mysql57"
+        self.mysql_port = 3306
```

Comment:
BTW maybe it worth using `get_open_port()` here too? (see #23462)

Reply:
> I've left get_open_port only for port mappings to localhost where it will clash.

OK, sounds reasonable (since default ip_local_port_range is 32k..65k)